File: TRClient/src/app/Entities/TaxWrapper.ts
import {TaxRemCAItem} from './TaxRemCAItem';
import {TaxItem}      from './TaxItem';
import {Awrapper}     from './Awrapper';

export class TaxWrapper extends Awrapper  {
 
    PeriodId: number = -1;
    Company: string = ''; 
    Vendor: string = '';

    PaymentMethod: string = ''; 
    DueDate: Date = new Date(1900,1,1); 
    JurisdictionCode: string = ''; 
    TaxId: string = ''; 
    DataEntryComplete: boolean = false; 
    PrevMthJourDiff: number = 0;
    DiffVsLastMthPaymnt: number = 0;
    ValidationComplete: number = 0;
    JournalDifference: number = 0;
    CalcVendorCommission: number = 0;
    CommissionNote: string = '';
    AvailTaxItems: TaxItem[] = [];
    AvailTaxRemCA: TaxRemCAItem[] = [];
 

    
 }

File: TRBusinessLayer/DataObjects/TaxWrapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace TRBusinessLayer.DataObjects
{
    public class TaxWrapper : AbstractDto
    {
        public int PeriodId { get; set; }
        public string Company { get; set; }
        public string Vendor { get; set; }
        public string PaymentMethod { get; set; }
        public DateTime DueDate { get; set; }
        public string JurisdictionCode { get; set; }
        public string TaxId { get; set; }
        public bool DataEntryComplete { get; set; }
        public bool ValidationComplete { get; set; }
        public bool JournalDifference { get; set; }
        public decimal PrevMthJourDiff { get; set; }
        public decimal DiffVsLastMthPaymnt { get; set; }
        
        public decimal CalcVendorCommission { get; set; }
        public string CommissionNote { get; set; }
        public List<TaxItem> AvailTaxItems { get; set; }       
        public List<TaxRemCA> AvailTaxRemCA { get; set; }
    }
}